src/Routing/SystemWebRoutingMiddleware.cs

```diff
@@ -0,0 +1,24 @@
+// MIT License.
+
+using System.Web.Routing;
+using Microsoft.AspNetCore.SystemWebAdapters.Features;
+
+namespace Microsoft.Extensions.DependencyInjection;
+
+internal sealed class SystemWebRoutingMiddleware(AspNetCore.Http.RequestDelegate next, RouteCollection routes)
+{
+    public Task InvokeAsync(HttpContextCore context)
```

Comment:
so if I understand, the objective is set with every invocation rather than on demand...